pkg/controller/namespace/namespace_controller.go

```diff
@@ -70,23 +73,24 @@ func NewNamespaceController(
 	// we found in practice though that some auth engines when encountering paths they don't know about may return a 50x.
 	// until we have verbs, we pre-populate resources that do not support list or delete for well-known apis rather than
 	// probing the server once in order to be told no.
-	opCache := operationNotSupportedCache{}
+	opCache := deletion.OperationNotSupportedCache{}
```

Comment:
I would prefer we keep opCache entirely as an internal piece of logic to the deleter object you defined and not have it at this scope.